app/controllers/calagator/sources_controller.rb

```diff
@@ -7,7 +7,11 @@ def import
     @importer = Source::Importer.new(params.permit![:source])
     respond_to do |format|
       if @importer.import
-        format.html { redirect_to events_path, flash: { success: render_to_string(layout: false) } }
+        if @importer.events.count == 1
+          format.html { redirect_to @importer.events.first, flash: { success: render_to_string(layout: false) } }
+        else
+          format.html { redirect_to events_path, flash: { success: render_to_string(layout: false) } }
```

Comment:
Quite a bit of duplication between these two lines. I suggest extracting a local variable for the redirect target to DRY this up.